packages/cohort/src/core.rs

```diff
@@ -332,17 +340,23 @@ impl Cohort {
             return Ok(());
         }
 
-        // Talos gave "go ahead"
+        // Check safepoint condition before installing ...
+
+        let safepoint = resp.safepoint.unwrap(); // this is safe for 'Committed'
```

Comment:
Same as previous comment regarding `panic`. 

Also we shouldn't have this logic here in the `transfer` method, because replicator calls the same method and this waiting doesn't apply for replicator.

Reply:
RE: panic. There is return statement on line 340

This is example usage coming from "generate_workload".  I don't think replicator calls this one. 
It calls database.batch  for now, but with [new PR](https://github.com/kindredgroup/talos/pull/41) it will call BatchExecutor::execute(Vec<StatemapItem>)